TECHMANIA/Assets/Scripts/Components/Main Menu/OptionsPanel.cs

```diff
@@ -381,17 +397,54 @@ public void OnTracksFolderBrowseButtonClick()
             MemoryToUI();
             Paths.ApplyCustomDataLocation();
         }
+#endif
     }
 
     public void OnSkinsFolderBrowseButtonClick()
     {
+#if UNITY_ANDROID
+        if (!Permission.HasUserAuthorizedPermission(Permission.ExternalStorageWrite))
+        {
+            Permission.RequestUserPermission(Permission.ExternalStorageWrite);
+        }
+        if (!Permission.HasUserAuthorizedPermission(Permission.ExternalStorageRead))
+        {
+            Permission.RequestUserPermission(Permission.ExternalStorageRead);
+        }
+        AndroidPlugin.OpenStorageFolder(gameObject.name, "OnAndroidSkinsFolderSelected", "", true);
+#else
         string[] folders = SFB.StandaloneFileBrowser
             .OpenFolderPanel("",
             Options.instance.skinsFolderLocation,
             multiselect: false);
         if (folders.Length == 1)
         {
-            Options.instance.skinsFolderLocation = folders[0];
+            Options.instance.tracksFolderLocation = folders[0];
+            MemoryToUI();
+            Paths.ApplyCustomDataLocation();
+        }
+#endif
+    }
+
+    private void OnAndroidTracksFolderSelected(string result)
+    {
+        if (result[0] == '{')
+        {
+            ContentInfo info = JsonUtility.FromJson<ContentInfo>(result);
+            Options.instance.tracksFolderLocation = info.path;
+            SelectTrackPanel.RemoveCachedLists();
```

Comment:
Please extract the common procedure in `OnTracksFolderBrowseButtonClick` and `OnAndroidTracksFolderSelected` into a separate method, called `OnTracksFolderSelected(string fullPath)`, for example. The same goes for skins.